core/src/core/pmmr/backend.rs

```diff
@@ -55,6 +55,9 @@ pub trait Backend<T: PMMRable> {
 	/// Iterator over current (unpruned, unremoved) leaf positions.
 	fn leaf_pos_iter(&self) -> Box<dyn Iterator<Item = u64> + '_>;
 
+	/// Number of leaves
+	fn n_leafs(&self) -> u64;
```

Comment:
We should think about a better name for this. Its not the number of leaves. Its the number of "not removed" leaf positions but that's a pretty awkward name.